webapp/src/app.tsx

```diff
@@ -2059,12 +2059,19 @@ export class ProjectView
                         });
                     }
                 }
+                pxt.tickEvent("deploy.start")
                 return pxt.commands.deployAsync(resp, {
                     reportDeviceNotFoundAsync: (docPath, compileResult) => this.showDeviceNotFoundDialogAsync(docPath, compileResult),
                     reportError: (e) => core.errorNotification(e),
                     showNotification: (msg) => core.infoNotification(msg)
                 })
+                    .then(() => {
+                        pxt.tickEvent("deploy.finished")
+                    }, () => {
+                        pxt.tickEvent("deploy.unfinished")
```

Comment:
Won't this prevent the catch clause below from ever executing?

Reply:
Does it? Shoot. I'll investigate

Reply:
Yup, you're right it definitely swallows all rejections. And catch() doesn't just get called for exceptions but any rejection so I can just remove the onreject logic here.